### PR TITLE
Test metrics

### LINKED ISSUE #9: Write unit tests for the multi-label metrics.
Write extensive unit tests for the multi-label metrics. We have to be sure that these work as expected.

### PR DESCRIPTION
**Pull request by JohnGiorgi**
# Overview

This PR adds the same suite of unit tests for `FBetaMeasureMultiLabel` as exist for its parent class `FBetaMeasure`.

## TODO

~~- [X] Make whatever changes are required so that a `mask` may be used.~~ (#13)
~~- [X] Make whatever changes are required to handle the case where a batch contains no instances of a given class.~~ (#14)

## Closes

Closes #9.

### CODE REVIEW

File: tests/training/metrics/test_fbeta_multi_label_measure.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+                [0, 0, 0, 0, 0],
+            ]
+        )
+
+        # detailed target state
+        self.pred_sum = [4, 3, 3, 4, 1]
+        self.true_sum = [4, 5, 2, 4, 0]
+        self.true_positive_sum = [3, 3, 2, 4, 0]
+        # I don't think this is correct for multi-label
+        self.true_negative_sum = [10.0, 10.0, 12.0, 11.0, 14.0]
```
Comment (MichalMalyska) [inline tests/training/metrics/test_fbeta_multi_label_measure.py:42]: Why are these floats ?

File: tests/training/metrics/test_fbeta_multi_label_measure.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        self.total_sum = [15, 15, 15, 15, 15]
+
+        # true_positive_sum / pred_sum
+        desired_precisions = [3 / 4, 3 / 3, 2 / 3, 4 / 4, 0 / 1]
+        # true_positive_sum / true_sum
+        desired_recalls = [3 / 4, 3 / 5, 2 / 2, 4 / 4, 0.00]
+        desired_fscores = [
```
Comment (MichalMalyska) [inline tests/training/metrics/test_fbeta_multi_label_measure.py:49]: This is the assumption that beta = 1. We can rename the metric to be F1measuremultilabel

Comment (JohnGiorgi) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:49]: Any reason to do that though? The code currently works with beta != 1. These unit tests assume beta is 1 yes, but so do Allen’s unit tests.

File: tests/training/metrics/test_fbeta_multi_label_measure.py (hunk 1 above)
Comment (JohnGiorgi) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:42]: I don’t remember but in any case they are wrong. I think to code that computes this comes from the parent class and works for the multi-class (not multi-label) setting.

File: tests/training/metrics/test_fbeta_multi_label_measure.py (hunk 2 above)
Comment (MichalMalyska) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:49]: yeah that's totally weird. I think this shouldn't be an issue since that's the case

File: tests/training/metrics/test_fbeta_multi_label_measure.py (hunk 1 above)
Comment (MichalMalyska) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:42]: yup, I think we never changed the true_negative_sum in our code.

Comment (JohnGiorgi) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:42]: Gotcha. I think for completeness sake we should update the method that computes `true_negative_sum` for the multi-label case.

**Comment by JohnGiorgi** [comment]
@MichalMalyska There is a block of code in the parent class (see [here](https://github.com/allenai/allennlp/blob/d73f8a91fc32d1a51235977fee2732aa61d7182c/allennlp/training/metrics/fbeta_measure.py#L139-L160)) that was dropped from our `FBetaMeasureMultiLabel`. It looks like it handles the situation where no document is labeled for a given class. Do you have any idea how this translates to the multi-label case? I don't really understand the concept of a "bin" here (several of their variables are named `*_bins`). If we can solve that, I can write the remaining 6 unit tests.

File: tests/training/metrics/test_fbeta_multi_label_measure.py (hunk 1 above)
Comment (MichalMalyska) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:42]: yeah for sure, I don't think it should be that difficult to do, and It's definitely a prereq if we want to merge this to Allen. We should still rename it to F1MeasureMultilabel or add the beta param option.

Comment (JohnGiorgi) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:42]: The beta param option does exist in our code, is that what you mean?

https://github.com/semantic-health/allennlp-multi-label/blob/f37fbc369cb1a30c3c62981844fab0acc877b433/allennlp_multi_label/training/metrics/fbeta_multi_class_measure.py#L64-L70

**Comment by MichalMalyska** [comment]
bincount just sums up the examples, but it has the edgecase that if input is all 0s then it returns an empty tensor. I don't really get their implementation without digging in too deep but it seems to just try to cover edge cases where predicitions are all zero or labels are all zero.

File: tests/training/metrics/test_fbeta_multi_label_measure.py (hunk 1 above)
Comment (MichalMalyska) [reply tests/training/metrics/test_fbeta_multi_label_measure.py:42]: Oh yeah I just checked it out, it should totally work without issue. I completely forgot I did that.

**Comment by JohnGiorgi** [comment]
I am going to merge this as is because it works (and has test coverage) for our intentend use case. I will open seperate issues to make this code work with masking/batches with no gold labels.
